addons/l10n_fr_certification/models/account.py

```diff
@@ -0,0 +1,155 @@
+# -*- coding: utf-8 -*-
+from hashlib import sha256
+from json import dumps
+
+from openerp import models, api, fields
+from openerp.tools.translate import _
+from openerp.exceptions import UserError
+
+ERR_MSG = _("According to the french law, you cannot modify a %s in order for its posted data to be updated or deleted. Field: %s")
+
+#forbidden fields
+MOVE_FIELDS = ['date', 'journal_id', 'company_id']
+LINE_FIELDS = ['debit', 'credit', 'account_id', 'move_id', 'partner_id']
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    l10n_fr_secure_sequence_number = fields.Integer(readonly=True)
+    l10n_fr_hash = fields.Char(readonly=True)
+    l10n_fr_string_to_hash = fields.Char(compute='_compute_string_to_hash', readonly=True, store=False)
+
+    def _get_new_hash(self, secure_seq_number):
+        """ Returns the hash to write on journal entries when they get posted"""
+        self.ensure_one()
+        #get the only one exact previous move in the securisation sequence
+        prev_move = self.search([('state', '=', 'posted'),
+                                 ('company_id', '=', self.company_id.id),
+                                 ('l10n_fr_secure_sequence_number', '=', int(secure_seq_number) - 1)])
+        if prev_move and len(prev_move) != 1:
+            raise UserError(
+               _('Error occured when computing the hash. Impossible to get the unique previous posted move'))
+
+        #build and return the hash
+        return self._compute_hash(prev_move.l10n_fr_hash if prev_move else '')
+
+    def _compute_hash(self, previous_hash):
+        """ Computes the hash of the browse_record given as self, based on the hash
+        of the previous record in the company's securisation sequence given as parameter"""
+        self.ensure_one()
+        hash_string = sha256(previous_hash + self.l10n_fr_string_to_hash)
+        return hash_string.hexdigest()
+
+    def _compute_string_to_hash(self):
+        def _getattrstring(obj, field_str):
+            field_value = obj[field_str]
+            if obj._fields[field_str].type == 'many2one':
+                field_value = field_value.id
+            return str(field_value)
+
+        for move in self:
+            values = {}
+            for field in MOVE_FIELDS:
+                values[field] = _getattrstring(move, field)
+
+            for line in move.line_ids:
+                for field in LINE_FIELDS:
+                    values[field] = _getattrstring(line, field)
```

Comment:
> if you have a PR, I'd gladly take it :+1:

Done. https://github.com/odoo/odoo/pull/18175

> FYI, we just removed that module in master (will be saas-17) because we need to review the restrictions and apply them only to the POS-part.

Do you mean features like in this PR ? https://github.com/OCA/l10n-france/pull/108
or other ? 

regards.